src/wazuh_db/wdb_upgrade.c

```diff
@@ -122,6 +133,12 @@ wdb_t * wdb_upgrade_global(wdb_t *wdb) {
         }
     }
 
+    if (post_upgrade) {
+        if (OS_SUCCESS != wdb_global_create_backup(wdb, output, "-post_upgrade")) {
```

Comment:
If the upgrade process fails, the database is already saved with `wdb_backup_global()`.
In that scenario, maybe we shouldn't create another backup of an empty database

Reply:
I have a suggestion for this
```
    if (version < updates_length) {
        if (OS_SUCCESS != wdb_global_create_backup(wdb, output, "-pre_upgrade")) {
            mwarn("Creating pre-upgrade global DB snapshot failed");
        }
        
        upgrade_success = TRUE;
        for (int i = version; i < updates_length; i++) {
            mdebug2("Updating database '%s' to version %d", wdb->id, i + 1);

            if (wdb_sql_exec(wdb, UPDATES[i]) == -1 || wdb_adjust_global_upgrade(wdb, i)) {
                mwarn("Failed to update global.db to version %d", i + 1);
                wdb = wdb_backup_global(wdb, version);
                upgrade_success = FALSE;
                break;
            }
        }
        
        if (upgrade_success) {
            if (OS_SUCCESS != wdb_global_create_backup(wdb, output, "-post_upgrade")) {
               mwarn("Creating post-upgrade global DB snapshot failed");
            }
        }
    }
```

Reply:
I like it!